saviSimulation/savi/jason_processing/WorldObject.java

```diff
@@ -7,50 +7,61 @@
 
-public class WorldObject{
-		//-----------------------------------------
-		// DATA (or state variables)
-		//-----------------------------------------
-		int ID;
-		protected PVector position;
-		String type;
-		SAVIWorld_model simulator;
-		PShape image;
-		int pixels;
-			//-----------------------------------------
-			// METHODS (functions that act on the data)
-			//-----------------------------------------
-			// Constructor: called when an object is created using
-			//              the "new" keyword. It's the only method
-			//              that doesn't have a type (not even void).
-		WorldObject(int id, PVector pos, int pixels, String Type, SAVIWorld_model sim, PShape image) {
-			// Initialize data values
-			ID = id;
-			position=pos;
-			type=Type;
-			simulator = sim;
-			this.image = image;
-			this.pixels = pixels;
-		}  
-		
-
-		public void draw(PVector position) {
-			simulator.stroke(0);
-
-			simulator.shapeMode(PConstants.CENTER);
-			simulator.shape(this.image, this.position.x, this.position.y,pixels,pixels);
-
-		}
-		
-		public void update(double simtime, double timestep, int perceptionDistance, int WIFI_PERCEPTION_DISTANCE,  List<WorldObject> objects, List<WifiAntenna> wifiParticipants) {
-			// most worldObjects do nothing
-		}
-
-
-		public PVector getPosition() {
-			// TODO Auto-generated method stub
-			return position;
-		}
+public class WorldObject {
+
+
+    //-----------------------------------------
+    // DATA (or state variables)
+    //-----------------------------------------
+    int ID;
+    protected PVector position;
+    String type;
+    SAVIWorld_model simulator;
+    PShape image;
+    int pixels;
+
+    //-----------------------------------------
+    // METHODS (functions that act on the data)
+    //-----------------------------------------
+    // Constructor: called when an object is created using
+    //              the "new" keyword. It's the only method
+    //              that doesn't have a type (not even void).
+    WorldObject(int id, PVector pos, int pixels, String Type, SAVIWorld_model sim, PShape image) {
+        // Initialize data values
+        ID = id;
+        position = pos;
+        type = Type;
+        simulator = sim;
+        this.image = image;
+        this.pixels = pixels;
+    }
+
+
+    public void draw(PVector position) {
+        simulator.stroke(0);
+
+        simulator.shapeMode(PConstants.CENTER);
+        simulator.shape(this.image, this.position.x, this.position.y, pixels, pixels);
+
+    }
+
+    public void update(double simtime, double timestep, int perceptionDistance, int WIFI_PERCEPTION_DISTANCE, List<WorldObject> objects, List<WifiAntenna> wifiParticipants) {
+        // most worldObjects do nothing
+    }
+
+
+    public PVector getPosition() {
+        // TODO Auto-generated method stub
+        return position;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public int getPixels() {
+        return pixels;
+    }
 
```

Comment:
Yup! Getters are added for type/pixels so that they can be accessed from the behaviour. This was required to be able to access from the new behaviour package. 

On a side note, it would be good to follow proper encapsulation protocol (private fields / public getters) so that we are not accessing fields directly. I can introduce this as a GitHub issue, and can be something we address separately.